src/Builder/DatasetBuilder.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace CodePrimer\Builder;
+
+use CodePrimer\Helper\BusinessModelHelper;
+use CodePrimer\Helper\FieldHelper;
+use CodePrimer\Model\BusinessBundle;
+use CodePrimer\Model\Dataset;
+use CodePrimer\Renderer\TemplateRenderer;
+use CodePrimer\Template\Template;
+use CodePrimer\Twig\LanguageTwigExtension;
+
+class DatasetBuilder implements ArtifactBuilder
+{
+    /**
+     * @return string[]
+     *
+     * @throws \Exception
+     */
+    public function build(BusinessBundle $businessBundle, Template $template, TemplateRenderer $renderer): array
+    {
+        $files = [];
+        foreach ($businessBundle->getDatasets() as $dataset) {
+            $files[] = $this->buildDataset($businessBundle, $dataset, $template, $renderer);
+        }
+
+        return $files;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function buildDataset(BusinessBundle $businessBundle, Dataset $dataset, Template $template, TemplateRenderer $renderer): string
+    {
+        $context = [
+            'bundle' => $businessBundle,
+            'subpackage' => 'Dataset',
+            'model' => $dataset,
+            'dataset' => $dataset,
```

Comment:
why passing both `model` and `dataset` here since it refers to the same object. You use these only for template generation, so maybe you can stick to `model` only.

Reply:
That is a good observation. Some templates uses the model variable to perform generic checks while others use the a specific one (dataset in this case). I would have to refactor how templates should be designed to set only one of them here and set the appropriate variable in the template